object/datastore.go

```diff
@@ -75,29 +77,24 @@ func (d Datastore) Path(path string) string {
 	return fmt.Sprintf("[%s] %s", name, path)
 }
 
-// URL for datastore access over HTTP
-func (d Datastore) URL(ctx context.Context, dc *Datacenter, path string) (*url.URL, error) {
-	var mdc mo.Datacenter
-	if err := dc.Properties(ctx, dc.Reference(), []string{"name"}, &mdc); err != nil {
-		return nil, err
-	}
-
-	var mds mo.Datastore
-	if err := d.Properties(ctx, d.Reference(), []string{"name"}, &mds); err != nil {
-		return nil, err
-	}
-
+// NewURL constructs a url.URL with the given file path for datastore access over HTTP.
+func (d Datastore) NewURL(path string) *url.URL {
```

Comment:
In general my preference/habit is to use pointer receivers, but the existing wrappers use value.